### PR TITLE
[IMP] payment, website_payment: readonly/invisible fields when provid…

### PR DESCRIPTION
**Pull request by niyasraphy**
…er is not installed

before this commit, from the payment providers menu, even though the provider is not installed in the database, the name, company, website and image fields are editable for the users.

after this commit, the fields will be editable/visible only after the provider is installed in the db.

publish/unpublish button will be shown only when provider is installed.

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/payment/views/payment_provider_views.xml [hunk 1]
```diff
@@ -48,11 +48,11 @@
                             </div>
                         </button>
                     </div>
-                    <field name="image_128" widget="image" class="oe_avatar"/>
-                    <widget name="web_ribbon" title="Disabled" bg_color="bg-danger" attrs="{'invisible': [('state', '!=', 'disabled')]}"/>
-                    <widget name="web_ribbon" title="Test Mode" bg_color="bg-warning" attrs="{'invisible': [('state', '!=', 'test')]}"/>
+                    <field name="image_128" widget="image" class="oe_avatar" attrs="{'readonly': [('module_state', '=', 'uninstalled')]}"/>
```
Comment (AntoineVDV) [inline addons/payment/views/payment_provider_views.xml:51]: Does this change anything? I tested on a runbot without your patch, and I can't update the image of an uninstalled provider even if it is _not_ readonly.

**Review by AntoineVDV — CHANGES_REQUESTED** [review]
Hello @niyasraphy, and thank you for your fix :)

May I ask you to make the following changes before merging the PR?
- Rebase your branch onto 16.0 and re-target the PR to 16.0. It will be automatically forward-ported up to master. You could use `git fetch origin 15.0 && git reset --hard origin/15.0 && git cherry-pick ae005141 && git push -f` to achieve that.
- Whenever possible, "factorize" the `attrs` that you add on multiple elements in the parent element.
For example, you add `attrs="{'invisible': [('module_state', '=', 'uninstalled')]}"` to both of the `action_toggle_is_published` buttons while you could add the `attrs` on the parent `<div class="oe_button_box" name="button_box">` element.
Another example is the `attrs` you add to the `website_payment` module. If you add the `attrs` on the parent `<group name="payment_state">` element, you can remove the `attrs` on `company_id`, `website_id`, and even on `code` and `state`.
In short: when it makes sense for all child elements to have the `attrs`, put it on the parent element.
- Resolve the comments below.
- Optional (if you feel like it): extend your patch to the fields that are still visible when browsing the "SEPA Direct Debit" payment provider record. I suspect that the `attrs` on the `<group name="payment_state">` element should also make its child elements invisible if `module_id` is not set, i.e. the module is in the odoo/enterprise repository.

File: addons/payment/views/payment_provider_views.xml (hunk 1 above)
Comment (niyasraphy) [reply addons/payment/views/payment_provider_views.xml:51]: currently the image is readonly as the banner is on top of it, once i apply the invisible attribute for the ribbon to display only when the module is installed, the image is getting editable even if module is not installed.

Comment (niyasraphy) [reply addons/payment/views/payment_provider_views.xml:51]: thanks for the feedback @AntoineVDV , as suggested i will update the PR and will let you know

**Comment by niyasraphy** [comment]
hi @AntoineVDV ,
thanks for the suggestions and feedbacks :heart: 

created new pr targeting  16.0:  https://github.com/odoo/odoo/pull/109703

as suggested moved the attributes to the parent level and simplified the code,  regarding the  last point about the "SEPA Direct Debit", i didn't get correctly, if you have time, can you clarify it for me.
thanks
